enhancements/kueue/openshift-kueue.md

```diff
@@ -0,0 +1,749 @@
+	FairSharing *kueuecfgv1beta1.FairSharing `json:"fairSharing,omitempty"`
+}
+
+// KueueStatus defines the observed state of Kueue
+type KueueStatus struct {
+	operatorv1.OperatorStatus `json:",inline"`
+}
+```
+
+We will use the kueue configuration [API](https://github.com/kubernetes-sigs/kueue/blob/main/apis/config/v1beta1/configuration_types.go).
+
+The operator reads the kueue configuration and generates a ConfigMap that the kueue manager deployment will use. 
+We will only expose a limited amount of configurations.
+
+The flow is as follows:
+
+1) User specifies a Kueue CRD
+1) Configuration is specified
+1) A Configmap will be created with the configuration specified
+1) Kueue deployment will use that configmap in the operand deployment.
+
+The operator will manage these components necessary for Kueue. 
+- Custom Resources
+- Cluster Roles
+- Cluster Role Bindings
+- Services
+- Mutating Webhook
+- Validating Webhook
+- ServiceMonitors
+- APIService for visibility
+- Roles
+- Certificates
+- Issuers
+
+The operator will create and apply these resources to the cluster.
+
+The operator has the following requirements.
+
+a. Konflux integration
+b. Support for x64 and ARM
+c. Disconnected
+d. FIPS
+
+The operator will be OLM managed and hosted on OperatorHub.
+
+### Workflow Description
+
+#### Batch Job Administrator
+
+As a admin, I want to enable Kueue to manage batch jobs.
+
+```yaml
+apiVersion: operator.openshift.io/v1beta1
+kind: Kueue
+metadata:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+   config:
+     integrations:
+       frameworks:
+         - "batch/job"
+```
+
+This will create a Kueue deployment that will manage batch jobs. 
+
+#### RHOAI Enablement
+
+Red Hat Openshift AI is already using Kueue in production. Their deployment can be replicated:
+
+```yaml
+apiVersion: operator.openshift.io/v1beta1
+kind: Kueue
+metadata:
+ labels:
+ name: cluster
+ namespace: openshift-kueue-operator
+spec:
+  config:
+     integrations:
+      frameworks:
+       - "batch/job"
+       - "kubeflow.org/mpijob"
+       - "ray.io/rayjob"
+       - "ray.io/raycluster"
+       - "jobset.x-k8s.io/jobset"
+       - "kubeflow.org/mxjob"
+       - "kubeflow.org/paddlejob"
+       - "kubeflow.org/pytorchjob"
+       - "kubeflow.org/tfjob"
+       - "kubeflow.org/xgboostjob"
+      externalFrameworks:
+       - "AppWrapper.v1beta2.workload.codeflare.dev"
+```
+
+RHOAI provides developer features and allows for enablement of more advanced features. 
+Due to this, it is requested that Kueue Operator will still allow the
+changing of feature gates to provide more advanced functionality.
+
+### API Extensions
+
+Our operator will have a Kueue CRD that will trigger the installation of Kueue.
+
+Kueue operand has the CRDs:
+
+- ClusterQueue
+- AdmissionChecks
+- LocalQueue
+- Workloads
+- ResourceFlavors
+- ProvisioningRequestConfigs
+- MultiKueueClusters
+- MultiKueueConfigs
```

Comment:
MK would be in TP, and FG would be disabled. Would we still be installing these CRDs by default?

Reply:
As of now I haven't really considered not installing CRDs.

We install all the CRDs but we could change that.

Reply:
It would be preferable to not install CRDs that are considered alpha on clusters that are not TechPreview.

If the upstream change the Alpha APIs in ways that are breaking (which they are entitled to do), it puts as at risk.

I'm be a strong vote for not installing these unless the cluster is tech preview

Reply:
Does RHOAI install these CRDs currently?

Reply:
I think so but they got broke by MultiKueue which was alpha at the time.

So there is some discussion to not install alpha APIs but atm all CRDs are installed.

Right @varshaprasad96 ?

Reply:
Yes, this was brought up, but changes haven't been made in kueue 0.10.x (RHOAI 2.18) yet. It was past the code freeze.

Reply:
updated to say we will remove alpha APIs.